network/local.go

```diff
@@ -0,0 +1,459 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"golang.org/x/net/context"
+	"reflect"
+	"sync"
+	"time"
+)
+
+// NewLocalRouter returns a fresh router which uses local connections. It takes
+// the default context.
+func NewLocalRouter(sid *ServerIdentity) (*Router, error) {
+	return NewLocalRouterWithContext(defaultLocalContext, sid)
+}
+
+// NewLocalRouterWithContext is the same as NewLocalRouter but takes a specific
+// LocalContext. This is useful to run parallel different local overlays.
+func NewLocalRouterWithContext(ctx *LocalContext, sid *ServerIdentity) (*Router, error) {
+	h, err := NewLocalHostWithContext(ctx, sid.Address)
+	if err != nil {
+		return nil, err
+	}
+	return NewRouter(sid, h), nil
+}
+
+// LocalContext keeps reference to all opened local connections
+// It also keeps tracks of who is "listening", so it's possible to mimics
+// Conn & Listener.
+type LocalContext struct {
```

Comment:
`LocalEnvironment` `LocalSetting` ? It does not only store queues, but do everything : create connections, pass messages around, setup listening functions etc..